### DISCUSSION TITLE
How to hide backend URLs in vercel.json from repo?

### DISCUSSION THREAD

**Discussion by MazzomesClone** [vercel/community#2684] [Help]
### Problem Description

<p>Hi! I have the following vercel.json file for my project:</p>
<code>{
    "rewrites": [
        {
            "source": "/api/:match*",
            "destination": "http://myapiurl.com/api/:match*"
        },
        {
            "source": "/public/:match*",
            "destination": "http://myapiurl.com/public/:match*"
        },
        {
            "source": "/(.*)",
            "destination": "/"
        }
    ]
}</code>

<p>I need this file and the URLs in the repo so it can be used by vercel, but ideally I want to hide the URLs, which would normally be done using environment variables.</p>
<p>Any insight as to how to go about this? I need the rewrites to make my authentication cookies work.</p>

**Comment by amyegan** [comment]
I think [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware#) could be a better solution for you. That will let you match paths based on conditional statements and rewrite to URLs stored in environment variables or [Edge Config](https://vercel.com/docs/storage/edge-config).

Example: https://vercel.com/docs/concepts/functions/edge-middleware/middleware-api#match-paths-based-on-conditional-statements
